### PR TITLE
Add a section for building and testing on the main readme

### PR DESCRIPTION
**Pull request by jsturtevant**
Adding some details on building the project and pre-requisites needed for running the tests. I stumbled upon all the pre-requisites as I started playing around with the project.  Hopefully this makes it a bit easier for the next person.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -443,3 +443,32 @@ wit`. You can link two components together with `wasm-tools compose` as well.
...
+```
+
+There are a few perquisites to testing the project. You only need the language compilers that you wish to run test against.
+
+- WASI SDK
+  - Download from wasi-sdk releases page. If you're using Windows, you need the one with mingw in its name.
+  - `curl -LO https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-20/wasi-sdk-20.0-linux.tar.gz`
+  - Create an environment variable called `WASI_SDK_PATH`` giving the path where you extracted the WASI SDK download, i.e., the directory containing `bin`/`lib`/`share`` folders.
+- Compilers for the target language:
+  - TinyGo - https://tinygo.org/
```
Comment (Mossaka) [inline README.md:462]: Can you add that both Go and TinyGo are needed? I don't remember the minimal version for TinyGo, but have been specified in Readme somewhere.

File: README.md [hunk 2]
```diff
@@ -443,3 +443,32 @@ wit`. You can link two components together with `wasm-tools compose` as well.
...
+  - Rust - wasi target: `rustup target add wasm32-wasi`
+  - Java - TeaVM-WASI `ci/download-teamvm.sh`
+  - C - [Clang](https://clang.llvm.org/)
+  - C# - [Dotnet 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
+
+To run tests for a single language:
```
Comment (Mossaka) [inline README.md:468]: To clarify, there are two suits of tests in the repo: codegen and runtime tests. This command specifically is just running rust runtime tests.

**Review by Mossaka — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jsturtevant) [reply README.md]: I missed the go dep since I had it installed already. Thanks!

File: README.md (hunk 2 above)
Comment (jsturtevant) [reply README.md]: Ok, This is what I was missing. Thanks for the help in the zulip chat too. I re-organized this and the testing doc now that I understand it more.

**Review by Mossaka — APPROVED** [review]

**Pull request merged**